File: main/src/main/java/com/sedmelluq/discord/lavaplayer/track/playback/AudioFrameRebuilder.java
package com.sedmelluq.discord.lavaplayer.track.playback;

/**
 * Interface for classes which can rebuild audio frames.
 */
public interface AudioFrameRebuilder {
  /**
   * Rebuilds a frame (for example by reencoding)
   * @param frame The audio frame
   * @return The new frame (may be the same as input)
   */
  AudioFrame rebuild(AudioFrame frame);
}

File: main/src/main/java/com/sedmelluq/discord/lavaplayer/track/playback/AudioFrameRebuilder.kt
package com.sedmelluq.discord.lavaplayer.track.playback

/**
 * Interface for classes which can rebuild audio frames.
 */
interface AudioFrameRebuilder {
    /**
     * Rebuilds a frame (for example by re-encoding)
     *
     * @param frame The audio frame
     * @return The new frame (might be the same as input)
     */
    fun rebuild(frame: AudioFrame): AudioFrame
}
